### PR TITLE
Tests for CreateService and annotations

### PR DESCRIPTION
**Pull request by cdrage**
Adds some tests for CreateService, specifically the initial generation
as well as providing an edge case when specifying "kompose.service.type"

### CODE REVIEW

**Comment by cdrage** [comment]
Fixes 1/2 of https://github.com/kubernetes-incubator/kompose/issues/242

**Comment by cdrage** [comment]
Still newbish to Go, so let me know what things I can improve on to make this more idiomatic!

File: pkg/transformer/kubernetes/k8sutils_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+	// An example object generated via k8s runtime.Objects()
+	kompose_object := kobject.KomposeObject{
+		ServiceConfigs: map[string]kobject.ServiceConfig{"app": service},
+	}
+	k := Kubernetes{}
+	objects := k.Transform(kompose_object, kobject.ConvertOptions{CreateD: true, Replicas: 3})
+
+	// Test the creation of the service
+	CreateService("foo", service, objects)
```
Comment (kadel) [inline pkg/transformer/kubernetes/k8sutils_test.go]: Shouldn't be there at least some kind of test that service has been created?
Maybe I just don't understand it, but it looks like this is not testing anything :-)

Comment (cdrage) [inline pkg/transformer/kubernetes/k8sutils_test.go]: @kadel you're right, i forgot to add the checking err in this test, was initially a copy-and-paste from the test below. thank you!

**Comment by cdrage** [comment]
@kadel me neither for the tests, I updated it to check that information is generated. I tried doing a comparison between port numbers (passing 123 and making sure we get 123 back). But I'm having a bit of difficulty digging through k8s code to find the correct way of doing it. Hopefully this update to the PR is sufficient!

**Comment by kadel** [comment]
there already is a function for comparing ports in [kubernetes_test.go](https://github.com/kubernetes-incubator/kompose/blob/master/pkg/transformer/kubernetes/kubernetes_test.go#L93)

**Comment by kadel** [comment]
you commited vim swap file ;-) -  `pkg/transformer/kubernetes/.k8sutils_test.go.swp`

**Comment by cdrage** [comment]
@kadel boo, that's why I added the .gitignore PR earlier xD okay, so i've updated this PR again with new code, although I'm not sure why the tests are erroring.

File: pkg/transformer/kubernetes/k8sutils_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	// An example object generated via k8s runtime.Objects()
+	kompose_object := kobject.KomposeObject{
+		ServiceConfigs: map[string]kobject.ServiceConfig{"app": service},
+	}
+	k := Kubernetes{}
+	objects := k.Transform(kompose_object, kobject.ConvertOptions{CreateD: true, Replicas: 3})
+
+	// Test the creation of the service with modified annotations (kompose.service.type)
+	svc := k.CreateService("foo", service, objects)
+	if svc.Spec.Type != api.ServiceType("NodePort") {
```
Comment (kadel) [inline pkg/transformer/kubernetes/k8sutils_test.go]: One last thing ;-)
You test only NodePort, it would be great to test also other options for service.type.
This might be great case for something like this: https://github.com/golang/go/wiki/TableDrivenTests

Comment (cdrage) [inline pkg/transformer/kubernetes/k8sutils_test.go]: @kadel sure, i'll work on that. i'll include it in the top test however since this test is specific to supplying nodeport (testing the annotations / labels :+1: )

Comment (cdrage) [inline pkg/transformer/kubernetes/k8sutils_test.go]: @kadel so I don't think I can compare anything else (`svc.Spec.Type` is blank, same as `svc.Spec.ClusterIP`, `svc.Spec.LoadBalancerIP`, etc.)

Should I just compare that it comes up as blank?

Comment (kadel) [inline pkg/transformer/kubernetes/k8sutils_test.go]: @cdrage sorry my comment might be little bit confusing. I should do it as line comment.

What I meant was that it might be nice to make this test more generic. Using table driven tests you can do test that will test all possible values for `kompose.service.type` in one test. Basicaly create more services with different `kompose.service.type` label

**Comment by cdrage** [comment]
@kadel updated with your code :) thanks man!

**Comment by kadel** [comment]
> @kadel updated with your code :) thanks man!

thanks ;-)
Sorry for confusion, sometimes it is easier for me to express myself in code than in english :laughing:

**Comment by cdrage** [comment]
@kadel 

01101110 01101111 00100000 01110000 01110010 01101111 01100010 01101100 01100101 01101101

:)

**Comment by kadel** [comment]
00111010 01000100
